reverse_replication/launcher.go

```diff
@@ -0,0 +1,273 @@
+		}
+	} else {
+		fmt.Println("Created changestream ", changeStreamName)
+	}
+
+	gcsclient, _ := storage.NewClient(ctx)
+	u, _ := url.Parse(sourceShardsFilePath)
+	rc, _ := gcsclient.Bucket(u.Host).Object(u.Path[1:]).NewReader(ctx)
+	bArr, _ := ioutil.ReadAll(rc)
+	rc.Close()
+	var data []interface{}
+	json.Unmarshal(bArr, &data)
+	arr := []string{}
+	for i := 0; i < len(data); i++ {
+		arr = append(arr, data[i].(map[string]interface{})["logicalShardId"].(string))
+	}
+
+	topicName := strings.Split(pubSubDataTopicId, "/")[3]
+	client, err := pubsub.NewClient(ctx, strings.Split(pubSubDataTopicId, "/")[1])
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer client.Close()
+	_, err = client.CreateTopic(ctx, topicName)
+	if err != nil {
+		if !(strings.Contains(err.Error(), "AlreadyExists")) {
+			fmt.Printf("could not create topic: %v\n", err)
+			return
+		} else {
+			fmt.Printf("topic '%s' already exists, skipping creation...\n", topicName)
+		}
+	} else {
+		fmt.Println("Created topic ", pubSubDataTopicId)
+	}
+
+	wg := &sync.WaitGroup{}
+	for i := 0; i < len(arr); i++ {
+		wg.Add(1)
+		go func(shardId string) {
+			defer wg.Done()
+			_, err := client.CreateSubscription(ctx, shardId, pubsub.SubscriptionConfig{
+				Topic:                 client.Topic(topicName),
+				AckDeadline:           600 * time.Second,
+				EnableMessageOrdering: true,
+				Filter:                fmt.Sprintf("attributes.shardId=\"%s\"", shardId),
+			})
+			if err != nil {
+				if !(strings.Contains(err.Error(), "AlreadyExists")) {
+					fmt.Printf("could not create subscription: %v\n", err)
+				} else {
+					fmt.Printf("subscription '%s' already exists, skipping creation...\n", shardId)
+				}
+				return
+			}
+			fmt.Println("Created sub: ", shardId)
+		}(arr[i])
+	}
+	wg.Wait()
+
+	c, err := dataflow.NewFlexTemplatesClient(ctx)
+	if err != nil {
+		fmt.Printf("could not create flex template client: %v\n", err)
+		return
+	}
+	defer c.Close()
+
+	launchParameters := &dataflowpb.LaunchFlexTemplateParameter{
+		JobName:  fmt.Sprintf("%s-ordering", jobNamePrefix),
+		Template: &dataflowpb.LaunchFlexTemplateParameter_ContainerSpecGcsPath{ContainerSpecGcsPath: ORDERING_TEMPLATE},
+		Parameters: map[string]string{
+			"changeStreamName":   changeStreamName,
+			"instanceId":         instanceId,
+			"databaseId":         dbName,
+			"spannerProjectId":   projectId,
+			"metadataInstance":   metadataInstance,
+			"metadataDatabase":   metadataDatabase,
+			"startTimestamp":     "",
+			"incrementInterval":  "10",
+			"sinkType":           "pubsub",
+			"pubSubDataTopicId":  pubSubDataTopicId,
+			"pubSubErrorTopicId": pubSubDataTopicId,
+			"pubSubEndpoint":     pubSubEndpoint,
+			"sessionFilePath":    sessionFilePath,
+		},
+		Environment: &dataflowpb.FlexTemplateRuntimeEnvironment{
+			NumWorkers:            5,
+			AdditionalExperiments: []string{"use_runner_v2"},
+			MachineType:           "n2-standard-4",
```

Comment:
This I think we should parameterise and default to n2-standard-4. Reason being - if performance is not as expected on the customer project, we could experiment with larger machines.